### PR TITLE
fix setup.py file

### PR DESCRIPTION
**Pull request by GreatBahram**

### CODE REVIEW

File: tir/tir.py [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-import lxml.html
 from collections import namedtuple
 
+import requests
 
-# named tuples:
-# used for parsed data and themes
+import lxml.html
 
-Date = namedtuple('Date', ['year'        # e.g. '2018'
```
Comment (Pouriya-Jahanbakhsh) [inline tir/tir.py]: For readability, I used a comment in front of each line. Here you have deleted it (`# e.g. '2018'`)!

File: tir/tir.py [hunk 2]
```diff
@@ -233,10 +233,9 @@ def __init__(self, tag, attr=None, value=None):
 class Request:
  """A class which makes request for fetching HTML page """
```
Comment (Pouriya-Jahanbakhsh) [inline tir/tir.py]: I agree this one. Do we need indentation for doc-string? I see just one space.

File: tir/tir.py [hunk 3]
```diff
@@ -233,10 +233,9 @@ def __init__(self, tag, attr=None, value=None):
...
     def __init__(self,
-                url='http://time.ir',
```
Comment (Pouriya-Jahanbakhsh) [inline tir/tir.py]: But I can't agree this, You removed default value of arguments. In wanted to add new flag (e.g. `-A` for adding custom user agents) and we could use that as parameter of this class. Here we can't!

File: setup.py [hunk 4]
```diff
@@ -1,31 +1,29 @@
 #!/usr/bin/env python3
-# -*- coding: utf-8 -*-
 
 import os
-import re
-import codecs
-from setuptools import setup, find_packages
 
+from setuptools import find_packages, setup
```
Comment (Pouriya-Jahanbakhsh) [inline setup.py]: Why did you change order of imported objects?

File: setup.py [hunk 5]
```diff
@@ -1,31 +1,29 @@
...
-     ,description=extract_metaitem('description')
-     ,author=extract_metaitem('author')
-     ,author_email=extract_metaitem('email')
-     ,maintainer=extract_metaitem('author')
-     ,maintainer_email=extract_metaitem('email')
-     ,platforms=['Any']
-     ,packages=find_packages()
-     ,install_requires=['requests', 'lxml']
-     ,setup_requires=['pytest-runner'])
+with open(os.path.join(here, 'tir', '__version__.py'), mode='rt', encoding='utf-8') as f:
```
Comment (Pouriya-Jahanbakhsh) [inline setup.py]: OMG, Such a nice trick !  
The bad news is that in `extract_metaitem` we are raising human-readable error if we can't find an item, But it can be done here too.
I learnt this approach of writing setup.py from @bear, specially from his [twitter library](https://github.com/bear/python-twitter/blob/master/setup.py). But your approach seems good too.

What about changing `__version__.py` to `__about__.py`?

**Comment by Pouriya-Jahanbakhsh** [comment]
Note about coding style :rofl:. Personally I prefer `comma-first style`. If you don't like it, You can change code-style of whole file **only if you change half of its main code** ! after that, that file is not mine anymore, that belongs to you.

After all, I really appreciate for your changes & ideas & etc.

File: tir/tir.py (hunk 1 above)
Comment (GreatBahram) [reply tir/tir.py]: it's obvious all people around the globe can understand meaning of year, you can do it in this way: 
```python
['year', # year which you know its meaning
'day', # which is obvious again
'name', 
]

File: tir/tir.py (hunk 2 above)
Comment (GreatBahram) [reply tir/tir.py]: yes, it seems I have missed that.

File: setup.py (hunk 5 above)
Comment (GreatBahram) [reply setup.py]: `__version__` is kind of standard way to do that. most famous packages used that as a rule of thumb. I prefer not to change that. but do whatever you want

File: tir/tir.py (hunk 1 above)
Comment (Pouriya-Jahanbakhsh) [reply tir/tir.py]: :rofl: the goal of comment is not explaining year, It's just a warn to remember that it's not integer, its string.

File: setup.py (hunk 4 above)
Comment (GreatBahram) [reply setup.py]: Because the order matters in PEP8 code style (first the builtins, then third parties and finally local pakcages). Python developers mostly use PEP8 code style I encourage you to follow that too it helps other people to read your code with ease instead of make them confuse

File: tir/tir.py [hunk 6]
```diff
@@ -257,19 +244,17 @@ def __init__(self
             headers['user-agent'] = user_agent
         self.headers = headers
 
-
     def get(self):
         request = requests.get(self.url, headers=self.headers)
         body = request.text
         assert(len(body) > 10240) # It's normals size is about 80K, but wee need at least 10K to process
```
[the remark this answers is not in the document]
Comment (Pouriya-Jahanbakhsh) [reply tir/tir.py]: agreed.

File: tir/tir.py (hunk 1 above)
Comment (GreatBahram) [reply tir/tir.py]: So use type hints

File: setup.py (hunk 4 above)
Comment (Pouriya-Jahanbakhsh) [reply setup.py]: couple years ago I saw that, but I did not use python for complete application anymore. Thanks, I think I need to follow the rules.

File: tir/tir.py (hunk 1 above)
Comment (GreatBahram) [reply tir/tir.py]: I'm not sure . But I think you can explain those by using doc-string like this one [link](https://stackoverflow.com/questions/1606436/adding-docstrings-to-namedtuples)
or maybe they don't need explanation because no-one gonna use those namedtuple except the developers, I'll think about it ...

Comment (Pouriya-Jahanbakhsh) [reply tir/tir.py]: So it's ok too. good job.

File: tir/tir.py [hunk 7]
```diff
@@ -166,19 +156,14 @@ def transform_month(month):
...
-def is_a_to_z(data):
-    for char in data:
-        number = ord(char)
-        if 96 < number < 123: # a-z
-            continue
-        if 64 < number < 91:  # A-Z
-            continue
-        return False
+def is_ascii_letters(string):
+    for character in string:
```
[the remark this answers is not in the document]
Comment (GreatBahram) [reply tir/tir.py]: Run this code in a Python REPL
```python
import this
```
it'll show you: 
"Readability counts." :-) my pleasure

File: tir/tir.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (Pouriya-Jahanbakhsh) [reply tir/tir.py]: What about the `assert`? can we use `ValueError` instead?

File: tir/tir.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (Pouriya-Jahanbakhsh) [reply tir/tir.py]: Yup, The zen of Python, Thanks for reminding.

File: tir/tir.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (GreatBahram) [reply tir/tir.py]: could you explain what's the purpose of that, please? I didn't get it. why 10240? you mean, if it has smaller size it's not OK?

File: tir/tir.py (hunk 1 above)
Comment (GreatBahram) [reply tir/tir.py]: Thanks

File: setup.py (hunk 4 above)
Comment (GreatBahram) [reply setup.py]: Your welcome bro, in your spare time read my blog posts. I usually write about Python things
https://virgool.io/@GreatBahram

File: tir/tir.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (Pouriya-Jahanbakhsh) [reply tir/tir.py]: In development, I fetched time.ir 100 times and saved contents in files. To know what different situations (at least in 100 calls) occurs. The bad thing that I saw is that sometimes it does not yield random quote! believe that?  
I saw that the normal size for each page is between 70-120 KB.  
Here we have 3 transformers which will be run on HTML contents, One for getting dates, One for getting calendar and one for getting random quote. I don't want to parse HTML and don't want transformers run when I know that something went wrong (e.g. it yields a short HTTP 504 page while they are updating, etc ...)

File: setup.py (hunk 4 above)
Comment (Pouriya-Jahanbakhsh) [reply setup.py]: Sure, on it.

File: tir/tir.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (GreatBahram) [reply tir/tir.py]: I'll test it and talk about it after that.
